packages/iov-tendermint-rpc/src/rpcclients/websocketclient.ts

```diff
@@ -77,8 +83,18 @@ export class WebsocketClient implements RpcStreamingClient {
     if (request.method !== "subscribe") {
       throw new Error(`Request method must be "subscribe" to start event listening`);
     }
-    const producer = new RpcEventProducer(request, this.socket, this.bridge);
-    return Stream.create(producer);
+
+    const query = (request.params as any).query;
+    if (typeof query !== "string") {
+      throw new Error("request.params.query must be a string");
+    }
+
+    if (!this.subscriptionStreams.has(query)) {
```

Comment:
And this is the core of the whole PR :)

Nicely done

Reply:
jap, thanks :) Great to have it deep deep down in the RPC client to make it work everywhere